### PR TITLE
Option to ignore all whitespaces.

### LINKED ISSUE #4248: Option to ignore all whitespaces
**Do you want to request a *feature* or report a *bug*?**

feature

**What is the expected behavior?**

Expose https://git-scm.com/docs/git-diff#git-diff---ignore-all-space through the GUI.

### PR DESCRIPTION
**Pull request by jbialobr**
Closes #4248.

Changes proposed in this pull request:
 - Add a new toolbar button and a context menu button to ignore all whitespace changes.
 
Screenshots before and after (if PR changes UI):
Before
qr

After
gitext

How did I test this code:
- Manually.

Has been tested on (remove any that don't apply):
 - GIT 2.15
 - Windows 10

### CODE REVIEW

**Comment by jbialobr** [comment]
>The tooltips also need to be localisable.

They are handled by the standard translation mechanism.

> Also please add the source icons to GitUI\Resources\Icons\originals and update _Icon Sources.txt with the source of the icons.

I created them from scratch.

>Perhaps we need to add better tooltips to them

I will change them.

File: GitUI/Editor/FileViewer.cs [hunk 1]
```diff
@@ -326,8 +333,15 @@ public string GetExtraDiffArguments()
...
-                diffArguments.Append(" --ignore-space-change");
+            if (ignoreAllWhitespaces.Checked)
+            {
+                diffArguments.Append(" --ignore-all-space");
+            }
+            else
+            {
+                if (IgnoreWhitespaceChanges)
+                    diffArguments.Append(" --ignore-space-change");
+            }
```
[the remark this answers is not in the document]
Comment (jbialobr) [reply GitUI/Editor/FileViewer.cs]: I don't see how - this is `else` branch.

[the remark this answers is not in the document]
Comment (RussKie) [reply GitUI/Editor/FileViewer.cs]: Apologies, was reading on a small factor device and somehow missed completely the args are in fact different.

**Comment by jbialobr** [comment]
The tooltips are changed.

**Pull request merged**